core/cmd/local_client_test.go

```diff
@@ -265,3 +272,157 @@ func TestClient_LogToDiskOptionDisablesAsExpected(t *testing.T) {
 		})
 	}
 }
+
+func TestClient_RebroadcastTransactions_WithinRange(t *testing.T) {
+	beginningNonce := uint(7)
+	endingNonce := uint(10)
+	gasPrice := big.NewInt(100000000000)
+	gasLimit := uint64(3000000)
+	set := flag.NewFlagSet("test", 0)
+	set.Bool("debug", true, "")
+	set.Uint("beginningNonce", beginningNonce, "")
+	set.Uint("endingNonce", endingNonce, "")
+	set.Uint64("gasPriceWei", gasPrice.Uint64(), "")
+	set.Uint64("gasLimit", gasLimit, "")
+	c := cli.NewContext(nil, set, nil)
+
+	tests := []struct {
+		dbName string
+		nonce  uint
+	}{
+		{"range_start", beginningNonce},
+		{"mid_range", beginningNonce + 1},
+		{"range_end", endingNonce},
+	}
+	for _, test := range tests {
+		t.Run(test.dbName, func(t *testing.T) {
+			// Use the a non-transactional db for this test because we need to
+			// test multiple connections to the datbase, and changes made within
+			// the transaction cannot be seen from another connection.
+			config, tempDB, cleanup := cltest.BootstrapThrowawayORM(t, fmt.Sprintf("rebroadcast_txs_%s", test.dbName))
+			defer cleanup()
+			require.NoError(t, tempDB.RawDB(func(db *gorm.DB) error { return migrations.Migrate(db) }))
```

Comment:
It might be nice to move the migration logic inside `BoostrapThroawayORM` and switch it on with a boolean flag or something because I imagine this might be a not uncommon use case.